src/error.rs

```diff
@@ -0,0 +1,24 @@
+use reqwest::{header::InvalidHeaderValue, StatusCode};
+use thiserror::Error;
+
+use crate::responses;
+
+#[derive(Error, Debug)]
+pub enum Error<Respone> {
+    #[error("encountered an error when performing an HTTP request")]
+    RequestError(#[from] reqwest::Error),
+    #[error("API responded with a client error: status code of {0}")]
+    ClientErrorResponse(StatusCode, Respone),
+    #[error("API responded with a server error: status code of {0}")]
+    ServerErrorResponse(StatusCode, Respone),
+    #[error("Health check failed: resource alarms are in effect")]
```

Comment:
Not all [health checks](https://www.rabbitmq.com/docs/monitoring#health-checks) are related to resources but IIRC all of them use the same HTTP status code for failures. So this message must be shortened but the enum value makes sense.